File: Merge Strings Alternately - Leetcode 1768/Merge Strings Alternately - Leetcode 1768.js
var mergeAlternately = function(word1, word2) {
    let A = word1.length, B = word2.length;
    let a = 0, b = 0;
    let result = '';

    while (a < A && b < B) {
        result += word1[a++];
        result += word2[b++];
    }

    while (a < A) {
        result += word1[a++];
    }

    while (b < B) {
        result += word2[b++];
    }

    return result;
};

File: Merge Strings Alternately - Leetcode 1768/Merge Strings Alternately - Leetcode 1768.go
package solution

func mergeAlternately(word1 string, word2 string) string {
    mergedWord := ""
    i:=0
    for i<len(word1) && i<len(word2){
        mergedWord += string(word1[i])+string(word2[i])
        i++
    }

    if i <len(word1){
        mergedWord += word1[i:]
    }

    if i <len(word2){
        mergedWord += word2[i:]
    }

    return mergedWord
}
